File: TrainingLog-api/TrainingLog.Domain/Entities/Race.cs
namespace TrainingLog.Domain.Entities
{
    public class Race
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime RaceDate { get; set; }
        public decimal DistanceMiles { get; set; }
        public TimeSpan? TargetTime { get; set; }

        // Foreign key
        public int TrainingPlanId { get; set; }
        public TrainingPlan? TrainingPlan { get; set; }
    }
}

File: traininglog-client/src/types/race.ts
export interface RaceDto {
  id: number;
  name: string;
  raceDate: string;
  distanceMiles: number;
  targetTime: string | null;
  trainingPlanId: number;
}

export interface CreateRaceDto {
  name: string;
  raceDate: string;
  distanceMiles: number;
  targetTime: string | null;
  trainingPlanId: number;
}
